engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
@vinaydt is running the tests on this PR to confirm that this PR is fixing the issue. So far the app is running for 19 hours without failing.